### PR TITLE
Refs [Task][41623] Handle Event: Search Screen

### PR DESCRIPTION
**Pull request by lanch1147**
### [Redmine link]
https://edu-redmine.sun-asterisk.vn/issues/41623

#### Screenshots (nếu có)
https://user-images.githubusercontent.com/86757854/134866857-83fd361d-592a-4105-be0b-af200cebdf7d.mov

### CODE REVIEW

File: Recipe Search/Recipe Search/Controllers/SearchViewController.swift [hunk 1]
```diff
@@ -95,6 +97,15 @@ final class SearchViewController: UIViewController {
         }
         return layout
     }
+    
+    @IBSegueAction func viewResultScreen(_ coder: NSCoder, sender: Any?) -> ResultViewController? {
+        let text = searchBar.text
+        let selectedCategories = sections[0].filters
+        let request = RecipeRequest(recipeName: text, categories: selectedCategories)
+        let tilte = "Results for \"\(text ?? "")\""
+        print(selectedCategories)
```
Comment (namtn-2711) [inline Recipe Search/Recipe Search/Controllers/SearchViewController.swift:106]: remove

File: Recipe Search/Recipe Search/Controllers/SearchViewController.swift [hunk 2]
```diff
@@ -138,3 +150,46 @@ extension SearchViewController: UICollectionViewDataSource {
...
 }
+
+extension SearchViewController: UICollectionViewDelegate {
+    func collectionView(_ collectionView: UICollectionView, shouldSelectItemAt indexPath: IndexPath) -> Bool {
+        return indexPath.section == 0 ? false : true
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+        let selectedItem = sections[indexPath.section].filters[indexPath.item]
+        sections[0].filters.append(selectedItem)
```
Comment (namtn-2711) [inline Recipe Search/Recipe Search/Controllers/SearchViewController.swift:161]:
```suggestion
        sections.first.filters.append(selectedItem)
```

File: Recipe Search/Recipe Search/Controllers/SearchViewController.swift [hunk 3]
```diff
@@ -138,3 +150,46 @@ extension SearchViewController: UICollectionViewDataSource {
...
+        let indexPath = IndexPath(item: sections[0].filters.count-1, section: 0)
```
Comment (namtn-2711) [inline Recipe Search/Recipe Search/Controllers/SearchViewController.swift:162]:
```suggestion
        let indexPath = IndexPath(item: sections.first.filters.count-1, section: 0)
```

File: Recipe Search/Recipe Search/Controllers/SearchViewController.swift (hunk 1 above)
Comment (tuanlvt-1117) [inline Recipe Search/Recipe Search/Controllers/SearchViewController.swift:106]: remove

File: Recipe Search/Recipe Search/Controllers/SearchViewController.swift [hunk 4]
```diff
@@ -138,3 +150,46 @@ extension SearchViewController: UICollectionViewDataSource {
         return header
     }
 }
+
+extension SearchViewController: UICollectionViewDelegate {
+    func collectionView(_ collectionView: UICollectionView, shouldSelectItemAt indexPath: IndexPath) -> Bool {
+        return indexPath.section == 0 ? false : true
```
Comment (tuanlvt-1117) [inline Recipe Search/Recipe Search/Controllers/SearchViewController.swift:156]:
```suggestion
        return indexPath.section != 0
```

Comment (lanch1147) [reply Recipe Search/Recipe Search/Controllers/SearchViewController.swift:156]: 😄 code nhiều quá bị ngáo anh ạ 😄 , để em sửa

File: Recipe Search/Recipe Search/Controllers/SearchViewController.swift [hunk 5]
```diff
@@ -138,3 +149,46 @@ extension SearchViewController: UICollectionViewDataSource {
...
+
+extension SearchViewController: UICollectionViewDelegate {
+    func collectionView(_ collectionView: UICollectionView, shouldSelectItemAt indexPath: IndexPath) -> Bool {
+        return indexPath.section != 0
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+        let selectedItem = sections[indexPath.section].filters[indexPath.item]
+        sections[0].filters.append(selectedItem)
+        let indexPath = IndexPath(item: sections[0].filters.count-1, section: 0)
```
Comment (namtn-2711) [inline Recipe Search/Recipe Search/Controllers/SearchViewController.swift:161]:
```suggestion
        let indexPath = IndexPath(item: sections[0].filters.count - 1, section: 0)
```

**Pull request merged**